test/functional/p2p_filter.py

```diff
@@ -174,6 +184,32 @@ def run_test(self):
         filter_node.send_and_ping(msg_filterload(data=b'', nHashFuncs=1))
         filter_node.send_and_ping(msg_filteradd(data=b'letstrytocrashthisnode'))
 
+    def run_test(self):
+        filter_peer = self.nodes[0].add_p2p_connection(FilterNode())
+        self.log.info('Test filter size limits')
+        self.test_size_limits(filter_peer)
+
+        self.log.info('Test BIP 37 for a node with fRelay = True (default)')
+        self.test_filter(filter_peer)
+        self.nodes[0].disconnect_p2ps()
+        wait_until(lambda: len(self.nodes[0].getpeerinfo()) == 0)
```

Comment:
in commit 5f0dd129a0:

It seems a burden on tests to call this to avoid races. It might be easy to forget and races might only appear long after the test has been written and merged. It seems less fragile to move this wait into the disconnect helper.

Reply:
Uh oh... I naively added a `wait_until(lambda: len(self.getpeerinfo()) == 0)` but `getpeerinfo` gets both p2p connections and connected nodes. I need to figure out a better way to do this.